app/models/katello/docker_manifest.rb

```diff
@@ -1,31 +1,42 @@
 module Katello
   class DockerManifest < Katello::Model
     include Concerns::PulpDatabaseUnit
-
-    has_many :docker_tags, :dependent => :destroy, :class_name => "Katello::DockerTag", :foreign_key => :docker_manifest_id
-    has_many :repository_docker_manifests, :dependent => :destroy
+    has_many :docker_tags, :as => :docker_taggable, :class_name => "Katello::DockerTag", :dependent => :destroy
+    has_many :repository_docker_manifests, :dependent => :destroy, :inverse_of => :docker_manifest
     has_many :repositories, :through => :repository_docker_manifests, :inverse_of => :docker_manifests
 
+    has_many :docker_manifest_list_manifests, :class_name => "Katello::DockerManifestListManifest",
+             :dependent => :delete_all, :inverse_of => :docker_manifest
+    has_many :docker_manifest_lists, :through => :docker_manifest_list_manifests, :inverse_of => :docker_manifests
+
     CONTENT_TYPE = Pulp::DockerManifest::CONTENT_TYPE
-    scoped_search :on => :name, :complete_value => true
-    scoped_search :relation => :docker_tags, :on => :name, :rename => :tag, :complete_value => true, :only_explicit => true
+    scoped_search :relation => :docker_tags, :on => :name, :rename => :tag, :complete_value => true
     scoped_search :on => :digest, :rename => :digest, :complete_value => true, :only_explicit => true
     scoped_search :on => :schema_version, :rename => :schema_version, :complete_value => true, :only_explicit => true
+    scoped_search :relation => :docker_manifest_lists, :on => :digest, :rename => :manifest_list_digest, :complete_value => true, :only_explicit => true
+
+    scope :orphaned, -> {
+      where("#{::Katello::DockerManifest.table_name}.id not in (select docker_manifest_id"\
+                  " from #{::Katello::DockerManifestListManifest.table_name})")
```

Comment:
I know this seems *more* complicated, but I imagine the current query might have performance issues if there are many docker manifest lists synced to the system.

```rb
joins("left outer join #{Katello::DockerManifestListManifest.table_name} on #{Katello::DockerManifest.table_name}.id = #{Katello::DockerManifestListManifest.table_name}.docker_manifest_id").where(Katello::DockerManifestListManifest.table_name => {id: nil})
```

Reply:
Actually I dont see orphaned being used.. I 'll just remove it ...